builder/dockerfile/typedcommand/commands.go

```diff
@@ -0,0 +1,316 @@
+package typedcommand
```

Comment:
In the docs we call these "instructions" not commands. It would be good to keep the terms consistent. Maybe we can name this package `instructions` ?